src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<Item>(
+  items: Item[],
+  key: keyof Item,
+): GroupsMap<Item> {
+  const result: GroupsMap<Item> = {};
+
+  for (const item of items) {
+    const value = String(item[key]);
```

Comment:
Consider checking if `item[key]` is undefined or null before converting it to a string. If `item[key]` is undefined or null, calling `String(item[key])` will result in the string 'undefined' or 'null', which might not be the desired behavior for grouping.